tools/fetch_aips

```diff
@@ -0,0 +1,140 @@
+
+        # Get storage service configuration
+        storage_service = StorageService.query.get(ss_id)
+
+        if storage_service is None:
+            logger.critical(f"Storage service ID {ss_id} not found.")
+            sys.exit(1)
+
+        # Create a fetch_job record in the AIPscan database and take note of its ID
+        datetime_obj_start = datetime.now().replace(microsecond=0)
+
+        fetch_job = database_helpers.create_fetch_job(
+            datetime_obj_start, session_id, ss_id
+        )
+        fetch_job_id = fetch_job.id
+
+        logger.info(f"Created fetch job record {fetch_job_id}.")
+
+        # Create directories, if need be
+        packages_dir = fetch.create_packages_directory(session_id)
+        fetch.create_mets_directory(session_id)
+        create_numbered_subdirs(session_id, package_list_no)
+
+        # Get package data
+        packages = fetch.get_packages(storage_service, packages_dir)
+
+        # Determine start and end package
+        total_packages = len(packages["objects"])
+
+        if page is None:
+            start_item = 1
+            end_item = total_packages
+        else:
+            start_item = ((page - 1) * packages_per_page) + 1
+            end_item = start_item + packages_per_page - 1
+
+        # Describe start and end package
+        if total_packages < end_item:
+            end_item = total_packages
+
+        # Make sure paging window is valid and delete fetch job if not
+        if start_item > total_packages:
+            logger.error(
+                f"Fetch job deleted. Page {page} would start at package {start_item} but there are only {total_packages} packages."
+            )
+            db.session.delete(fetch_job)
+            db.session.commit()
+            sys.exit(1)
+
+        logger.info(
+            f"Processing packages {start_item} to {end_item} of {total_packages}."
+        )
+
+        # Import packages
+        api_url = fetch.assemble_api_url_dict(storage_service)
+
+        processed_packages = fetch.import_packages(
+            packages,
+            start_item,
+            end_item,
+            api_url,
+            ss_id,
+            session_id,
+            package_list_no,
+            fetch_job_id,
+            packages_per_page,
+            logger,
+        )
+
+        fetch_job = database_helpers.update_fetch_job(
+            fetch_job_id, processed_packages, total_packages
+        )
+
+        summary = "aips: '{}'; sips: '{}'; dips: '{}'; deleted: '{}'; replicated: '{}'".format(
+            fetch_job.total_aips,
+            fetch_job.total_sips,
+            fetch_job.total_dips,
+            fetch_job.total_deleted_aips,
+            fetch_job.total_replicas,
+        )
+        logger.info("%s", summary)
+        logger.info(
+            f"Updated fetch job record {fetch_job_id} with package type counts."
+        )
+        logger.info("Processing complete.")
+
+
+if __name__ == "__main__":
+    sys.exit(main())
```

Comment:
Looks like main is doing some extra work because it's a click.command, seehttps://click.palletsprojects.com/en/8.1.x/exceptions/. It looks from some examples I've seen that `sys.exit` might not be necessary in this context.